### ISSUE TITLE
Option to ignore arrow functions in `consistent-function-scoping`

### ISSUE THREAD

**Issue by garyking** [sindresorhus/eslint-plugin-unicorn#596]
Please allow the option to ignore arrow functions in the `consistent-function-scoping` rule. Sometimes I create a one liner function, and I don't want to have to move it to the highest scope.

**Comment by fisker** [comment]
Any real world case?

**Comment by garyking** [comment]
I enabled this rule, and got dozens of warnings, so I do have some examples. Many are legitimate warnings, but some are for one-liner arrow functions that I don't feel like it makes sense to extract them. 

One example, for an NBA app, written in TypeScript; this function returns a formatted lead for either the away or home team. For the team that's winning in a live game, it shows `(+5)` to indicate that team is winning by 5 points. This code is part of a larger set of exported functions, from one object. 

The line that gets reported as an error is `const formatLead = (lead: number): string => `(+${lead})`;`, which I feel is such a short function, and one that pertains specifically to this function, that it shouldn't be extracted. And in this particular case, if I extracted that function, then it would be moved far away from the `getTeamLeads` function. Let's say `getTeamLeads` is on line 500, and `export default` is on line 10, then I'd have to move `formatLead` to line 5 or so (since I don't want to add it to the `export default` object).

```typescript
export default {
  // There are several more functions in this exported object, as indicated by
  // the `this.` calls to other functions.
  
  getTeamLeads(
    game: nba.games.FormattedGame,
  ): { awayLead: string; homeLead: string } {
    const winningTeam = this.getWinningTeam(game);

    // Score lead
    const scoreDifference = this.getScoreDifference(game);
    let awayLead: number | null = null;
    let homeLead: number | null = null;

    // Show the score lead for the winning team.
    if (winningTeam === 'away') {
      awayLead = scoreDifference;
    } else if (winningTeam === 'home') {
      homeLead = scoreDifference;

      // Both teams are tied.
    } else {
      awayLead = 0;
      homeLead = awayLead;
    }

    const formatLead = (lead: number): string => `(+${lead})`;
    const awayLeadFormatted = _.isNumber(awayLead) ? formatLead(awayLead) : '';
    const homeLeadFormatted = _.isNumber(homeLead) ? formatLead(homeLead) : '';

    return { awayLead: awayLeadFormatted, homeLead: homeLeadFormatted };
  },
}
```

**Comment by garyking** [comment]
Another example I have is this piece of code, which is used to show the Git branches of a project. The final line, accepts two functions. I've split the functions off, rather than passing them directly, since I think it's easier to read this way. This rule reports errors on both functions.

```typescript
export default {
  async gitBranch(): Promise<void> {
    const getDataCallback = async (git: Git): Promise<GitBranch[] | null> =>
      await git.branches();

    const logCallback = (branches: GitBranch[]): string => {
      const messages = branches.map((branch) => {
        if (branch.isCurrent) {
          return chalk`{green ${branch.name}}`;
        }

        return branch.name;
      });

      return messages.join('\n');
    };

    await gitUtils.printGitData({ getDataCallback, logCallback });
  },
}
```

**Comment by fisker** [comment]
Sorry for the late response, I found that, ignore arrow functions doesn't make sense in your cases.

The first one,

```js
const formatLead = (lead: number): string => `(+${lead})`;
```

It should move to the outer scope, because it a pure function, why 

```js
function formatLead(string ) {
  return `(+${lead})`;
}
```

should report, and the arrow one should not? There is no difference.

The `getDataCallback` and `logCallback` are the same.

**Comment by garyking** [comment]
Okay. I will keep the rule disabled for myself then, because I feel like these functions shouldn't be moved out, since they are only used for one function.

**Comment by fisker** [comment]
In all cases, the function is only used in one function, isn't right? If other function used it, it can't live in  the inner scope.

**Comment by garyking** [comment]
Yeah, I use arrow functions for functions that are only used in one function, especially if it's just a one-liner. To try and keep the code cleaner.

**Comment by fisker** [comment]
Personally, I don't treat them differently, but I guess we could have option to ignore them.

/cc @sindresorhus

**Comment by sindresorhus** [comment]
Same here. Not something I would use. But I guess we could add a option to ignore **inline** arrow functions.

**Comment by Arttse** [comment]
How about this simple example with Redux Thunk?

Arrow function (throws error):
```js
// ESLint: Move arrow function to the outer scope.(unicorn/consistent-function-scoping)
//
//                              here
//                              ⬇
export const someAction = () => (dispatch) =>
  dispatch({
    type: SOME_TYPE,
  });
```

Function declaration (no error):
```js
export function someAction() {
  return (dispatch) =>
    dispatch({
      type: SOME_TYPE,
    });
}
```

Is it bug or feature? Is it normal or not?

**Comment by egemon** [comment]
@fisker same issue with recompose

**Comment by fisker** [comment]
Somehow, I missed this.

@Arttse It's bug, they all should reported.

**Comment by fisker** [comment]
I took a quick look, it's caused by `scopeManager.acquire` it can't get scope, not sure how to fix it now.
